src/libraries/System.Private.CoreLib/src/System/ParamsArray.cs

```diff
@@ -1,75 +1,39 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+// These types are temporary workarounds for an inability to stackalloc object references.
+// Once we're able to do `stackalloc object[n]`, these can be removed.
+
+// Suppress warnings for unused private fields
+#pragma warning disable CS0169
+#pragma warning disable CA1823
+#pragma warning disable IDE0051
+
 namespace System
 {
-    internal readonly struct ParamsArray
+    internal struct TwoObjects
```

Comment:
Why not just use a value tuple?

Reply:
Primarily that ValueTuple brings with it a whole bunch of stuff that would then need to be kept if string formatting was used in a trimmed app but `(ref, ref)` or `(ref, ref, ref)` weren't otherwise, and secondarily that this is already borderline in relying on a layout that's technically not guaranteed but we know it'll be ok and relying on ValueTuple's layout even further felt... icky.  That said, I don't have a strong objection to changing it to `ValueTuple<...>`.

Reply:
No strong preference, was mostly just wondering as I thought it would save a few lines.

Reply:
Ok